### ISSUE TITLE
Name collision in `combine` with `domain`

### ISSUE THREAD

**Issue by Anber** [effector/effector#1064]
**What is the current behavior:**

The current behaviour is `undefined is not an object (evaluating 's.domain.hooks.store')`.

The reproduction case https://share.effector.dev/26YH41BQ:
```ts
const $domain = createStore(null);  
const $combined = combine({ domain: $domain });
```

**Which versions of effector packages, and which browser and OS are affected by this issue? Did this work in previous versions of effector?:**

effector@23.2.0. In 22.8.3, everything was fine.

**Comment by sergeysova** [comment]
`domain` is the specific name to set domain for stores made by `combine`.

Your workaround looks like this:

```ts
const $combined = combine($domain, (domain) => ({ domain }))
```

**Comment by Anber** [comment]
Hi @sergeysova!

Yep, that's how I resolved it. It's just strange behaviour. In my case, `combine` gets a dynamic object, and it was a huge surprise that there are forbidden keys, especially when there was no such thing in the previous version. At least a readable error would be helpful.

**Comment by zerobias** [comment]
Fixed in effector 23.2.1
